src/recap

```diff
@@ -146,492 +134,440 @@ print_usage() {
-	fi
-
-	# print the sar report to the resources file
-	echo "SAR$FLAGS report" >> $RESOURCES_FILE
-	sar$FLAGS >> $RESOURCES_FILE
-}
-
-# print the output of "apachectl fullstatus" to the resources file
+  local OPTION=$1
+  local LOGFILE=$2
+  local FLAGS=''
+  # check to see if we're going to use any parameters for sar
+  if [[ ${OPTION} == "r" ]]; then
+    FLAGS=" -r"
+  elif [[ ${OPTION} == "q" ]]; then
+    FLAGS=" -q"
+  else
+    FLAGS=""
+  fi
+
+  echo "SAR${FLAGS} report" >> "${LOGFILE}"
+  sar${FLAGS} >> "${LOGFILE}"
+}
+
+# print the output of "apachectl fullstatus" to the specified file
 print_httpd_fullstatus() {
-	echo "Apache Status report" >> $RESOURCES_FILE
-	apachectl fullstatus >> $RESOURCES_FILE
+  local LOGFILE=$1
+  echo "Apache Status report" >> "${LOGFILE}"
+  apachectl fullstatus >> "${LOGFILE}"
 }
 
-# print the output of "netstat -ntulpae" to the netstat file
+# print the output of "netstat -ntulpae" to the specified file
 print_netstat() {
-	echo "Network connections" >> $NETSTAT_FILE
-	netstat $OPTS_NETSTAT >> $NETSTAT_FILE
+  local LOGFILE=$1
+  echo "Network connections" >> "${LOGFILE}"
+  netstat ${OPTS_NETSTAT} >> "${LOGFILE}"
 }
 
-# print the output of "netstat -s" to the netstat file
+# print the output of "netstat -s" to the specified file
 print_netstat_sum() {
-	echo "Network traffic summary" >> $NETSTAT_FILE
-	netstat $OPTS_NETSTAT_SUM >> $NETSTAT_FILE
+  local LOGFILE=$1
+  echo "Network traffic summary" >> "${LOGFILE}"
+  netstat ${OPTS_NETSTAT_SUM} >> "${LOGFILE}"
 }
 
 # print the output of "mysqladmin status" to the mysql file
 print_mysql() {
-        # We only support a single DOTMYDOTCNF for MySQL when running Plesk
-        if [ -f /etc/psa/.psa.shadow ]
-        then
-                echo "MySQL status" >> $MYSQL_FILE
-                mysqladmin -uadmin -p`cat /etc/psa/.psa.shadow` status >> $MYSQL_FILE
-                print_blankline $MYSQL_FILE
-        else
-                for cnf in $( sed -e 's/,/ /g' <<<"${DOTMYDOTCNF}" )
-                do
-                        if [ -f $cnf ]
-                        then
-                                echo "MySQL $cnf status" >> $MYSQL_FILE
-                                mysqladmin --defaults-file=$cnf status >> $MYSQL_FILE
-                                print_blankline $MYSQL_FILE
-                        else
-                                echo "MySQL $cnf for status does not exist. Please check DOTMYDOTCNF in /etc/recap" >> $MYSQL_FILE
-                                print_blankline $MYSQL_FILE
-                        fi
-                done
-        fi
+  local LOGFILE=$1
+  # We only support a single DOTMYDOTCNF for MySQL when running Plesk
+  local PLESK_FILE="/etc/psa/.psa.shadow"
+  if [[ -r "${PLESK_FILE}" ]]; then
+    echo "MySQL status" >> "${LOGFILE}"
+    mysqladmin -uadmin -p$(cat "${PLESK_FILE}") status >> "${LOGFILE}"
+    print_blankline "${LOGFILE}"
+  else
+    for cnf in $( sed -e 's/,/ /g' <<<"${DOTMYDOTCNF}" ); do
```

Comment:
Ah, I see your point now, that seems to be valid, somehow unusual but valid as any file can have white spaces, I'd definitely cover this use case as long as mysql does it as well via its config file, and it does:
```bash
[root@gnu ~]# mysqladmin --defaults-extra-file="/root/my config.cnf" ping
mysqld is alive
```

In that case we will need to move to support only comma separated values and not white spaces, considering the white spaces are part of a name, e.g.

* Valid config:
  - DOTMYDOTCNF="/root/.my.cnf,/root/.my2.cnf,/some/path with/spaces in name.cnf"
  - DOTMYDOTCNF="  /some/path/my.cnf , /some/other/my.cnf "  (in this case we can remove trailing spaces)
* Invalid config:
  - DOTMYDOTCNF="/some/path/my.cnf /some/other/my.cnf"

Or not support it at all.